common/extended_window.hpp

```diff
@@ -176,5 +182,13 @@ namespace ILLIXR {
             */
 #endif
         }
+
+        ~xlib_gl_extended_window() {
+            XDestroyWindow(dpy, win);
+            Window root = DefaultRootWindow(dpy);
+            XDestroyWindow(dpy, root);
+            //glXDestroyContext(dpy, glc);
```

Comment:
Comment this in, and also bring the [_entire_ destructor](https://github.com/ILLIXR/ILLIXR/blob/master/timewarp_gl/plugin.cpp#L783) of timewarp here -- timewarp shouldn't be cleaning up resources allocated by this service.
The only thing I'm unsure of is whether the color map needs to be freed explicitly or not, given that the [XCloseDisplay](https://tronche.com/gui/x/xlib/display/XCloseDisplay.html) documentation says that it destroys all the resources attached to the display.

Reply:
Ah that makes sense. I actually commented out `glxDestroyContext` because I wasn't able to get it to run without segfaulting. Maybe moving the `timewarp_gl` destructor here will help?

Reply:
Yup, timewarp was the culprit. It already freed the context, resulting in the segfault here. Also remember to make the context current here before doing anything with it (it's already taken care of in the timewarp desctructor).